core/src/sumeragi/network_topology.rs

```diff
@@ -181,7 +188,23 @@ impl Topology {
         view_change_index: u64,
         new_peers: UniqueVec<PeerId>,
     ) -> Self {
-        let mut topology = Topology::new(block.commit_topology().clone());
+        let created_with_vrf_state = block.header().vrf_state.clone();
+        let mut rng = StdRng::seed_from_u64(u64::from_le_bytes(
+            HashOf::new(&created_with_vrf_state).as_ref()[0..8]
+                .try_into()
+                .expect("Cannot fail"),
+        ));
+
+        let mut topology = {
+            let mut shuffle_peers: Vec<PeerId> = block.commit_topology().clone().into();
```

Comment:
is it really a necessity that we store `commit_topology` in the block? Can't we just use the current topology of sumeragi? Because this field can blow the chain size up significantly

Reply:
It is. We could imagine some scheme where we store  only a delta and can infer the current topology. But I think it's probably fine since the block headers don't stay in memory. Once 2 blocks have passed this data should only be on disk. I say the chance of bugs or mistakes is a lot higher if we try to be smart here. Does not seem worth it.

Reply:
> But I think it's probably fine since the block headers don't stay in memory

that's true

> We could imagine some scheme where we store only a delta and can infer the current topology

can't we replay topology while doing block replay? I'm saying it will be computed anyhow, so why store it in the block? 

> I say the chance of bugs or mistakes is a lot higher if we try to be smart here. Does not seem worth it

I need a better argument why it's necessary to store it in the block header

Reply:
It isn't necessary, you are right that we could replay the topology with the block replay. It didn't seem worth it to me since the topology is quite small. But if you think we should replay it then I could work on that.

Reply:
@mversic I misspoke in our call. We would need to sort after we make this change and keep track of Peers without storing it in the block header. Right now we use the commit topology and that will be the same for all peers since it is on the block header. Once we remove the topology from the header we would need track the peers with a UniqueVec for consistent shuffling on different peers. But as of now there is no thing I needed to force push to the PR.